### PR TITLE
feat: dapp & verify credentials documentation

### PR DESCRIPTION
**Pull request by usmangeek**
Please review the changes. This PR includes the documentation for verification flow using credential API.

### CODE REVIEW

File: docs/develop/05_dApp/01_welcome.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (tjwelde) [inline docs/develop/05_dApp/01_welcome.md]: This empty screen seems unnecesary

File: docs/develop/05_dApp/verifier/01_session.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+id: session
+title: Session
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import Session from '!!raw-loader!@site/code_examples/workshop/verifier/session.ts';
+
+
+In order to create verifier, you must define session. Following code will help you create a  definition of session. The session is required for the communication between the dApp and the extension.
```
Comment (tjwelde) [inline docs/develop/05_dApp/verifier/01_session.md]:
```suggestion
In order to create a Verifier application, you must define a session. Following code will help you create the definition of a session. The session is required for the communication between the dApp and the extension.
```

File: docs/develop/05_dApp/verifier/02_requestCredential.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+id: request
+title: Request Credential
```
Comment (tjwelde) [inline docs/develop/05_dApp/verifier/02_requestCredential.md]:
```suggestion
title: Requesting a  Credential
```

File: docs/develop/05_dApp/verifier/02_requestCredential.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import RequestCredentials from '!!raw-loader!@site/code_examples/workshop/verifier/requestCredential.ts';
+
+Request has to be sent from the frontend with ctype and session ID. This code block will add challenge, ctype hash and the message body type to the request. The updated request is then sent to the extension. This example uses email ctype. Many other ctypes could also be used.
```
Comment (tjwelde) [inline docs/develop/05_dApp/verifier/02_requestCredential.md]:
```suggestion
To request a credential, a message has to be sent from the dApp to the extension with ctype and session ID. The following code block shows, how to add a challenge, ctype hash and the message body type to the request message. The updated request is then sent to the extension. This example uses the socialkyc email ctype. Many other ctypes could also be used.
```

File: docs/develop/05_dApp/verifier/03_verifyCrendentials.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+id: verify
+title: Verify Credential
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import VerifyCredentials from '!!raw-loader!@site/code_examples/workshop/verifier/verifyCredentials.ts';
+
+Request should be sent from the frontend with message from extension and session ID. This code block will use <a href="https://kiltprotocol.github.io/sdk-js/modules/_kiltprotocol_messaging.html" target="_blank"> Decrypt Messages</a> protocols from KILT SDK and returns the verified credentials. This information can be displayed through the frontend.
```
Comment (tjwelde) [inline docs/develop/05_dApp/verifier/03_verifyCrendentials.md]: Is this about receiving a credential from an extension?

File: docs/develop/05_dApp/verifier/04_sendRequest.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+id: send
+title: Send Request
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import SendRequest from '!!raw-loader!@site/code_examples/workshop/verifier/sendRequest.ts';
+
+Credential Request should be sent from the frontend to extension. Extension sends back the requested credential. That credential is then sent for verification to the backend.
```
Comment (tjwelde) [inline docs/develop/05_dApp/verifier/04_sendRequest.md]: is this about how to send a request in general?

**Review by tjwelde — CHANGES_REQUESTED** [review]
Unfortunately, this is not enough to understand how to make a verifier dApp.
I recommend an overview document, which first explains the overall process and than going into more detail in separate pages.

File: docs/develop/05_dApp/01_welcome.md (hunk 1 above)
Comment (usmangeek) [reply docs/develop/05_dApp/01_welcome.md]: I think we can utilise it for the overview.

File: docs/develop/05_dApp/verifier/03_verifyCrendentials.md (hunk 5 above)
Comment (usmangeek) [reply docs/develop/05_dApp/verifier/03_verifyCrendentials.md]: Yes

File: docs/develop/05_dApp/verifier/04_sendRequest.md (hunk 6 above)
Comment (usmangeek) [reply docs/develop/05_dApp/verifier/04_sendRequest.md]: Yes, its more about sending from frontend.

**Comment by tjwelde** [comment]
Please also format the code examples

File: code_examples/workshop/verifier/session.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+export async function getSession(
+  provider: InjectedWindowProvider,
+  wallet: string
+): Promise<Session> {
+  if (!provider) {
+    throw new Error('No provider')
+  }
+
+  window.sessionStorage.setItem('wallet', wallet)
```
Comment (tjwelde) [inline code_examples/workshop/verifier/session.ts]: this seems to be unnecessary for normal usecases

File: code_examples/workshop/verifier/session.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  const dAppName = 'Your dApp Name '
+
+  const session = await provider.startSession(
+    dAppName,
+    dAppEncryptionKeyUri, // from backend
+    challenge // from backend
+  )
+
+  async function send(message: IEncryptedMessage): Promise<void> {
+    message.receiverKeyId = message.receiverKeyUri
```
Comment (tjwelde) [inline code_examples/workshop/verifier/session.ts]: this is just needed for backwards compatibility, so I think it is not needed for new projects. same with the stuff in listen.

File: docs/develop/05_dApp/verifier/03_verifyCrendentials.md (hunk 5 above)
Comment (tjwelde) [reply docs/develop/05_dApp/verifier/03_verifyCrendentials.md]: should be clarified. Is this the  backend code here?

File: docs/develop/05_dApp/verifier/04_sendRequest.md (hunk 6 above)
Comment (tjwelde) [reply docs/develop/05_dApp/verifier/04_sendRequest.md]: So the frontend might send a message, but the backend does too? How does the frontend send a message. They are encrypted, right?

Comment (usmangeek) [reply docs/develop/05_dApp/verifier/04_sendRequest.md]: Yes

File: docs/develop/05_dApp/02_well-known-did-config.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+## dApp Steps
+
+Add the following URI to your dApp's root. The extension will make an HTTP GET request to this URI, and your dApp will respond with the domain configuration resource.
+
+`/.well-known/did-configuration.json`
+
+### Attesting the Domain Linkage Credential
+
+Upon receiving the GET request from the extension, your dApp will make a claim that contains its domain origin and DID. Then it will self-attest the claim. This self-attestation is referred to as a Domain Linkage Credential.
```
Comment (tjwelde) [inline docs/develop/05_dApp/02_well-known-did-config.md]: is this done ad-hoc, or do I have to prepare it beforehand?

**Review by tjwelde — COMMENTED** [review]
The well-known-did-config documentation looks pretty good. just a question

File: code_examples/workshop/verifier/session.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    dAppName,
+    dAppEncryptionKeyUri, // from backend
+    challenge // from backend
+  )
+
+  async function send(message: IEncryptedMessage): Promise<void> {
+    return session.send(message)
+  }
+
+  async function listen(
```
Comment (tjwelde) [inline code_examples/workshop/verifier/session.ts]: the `send` and `listen` function are now not necessary anymore

File: docs/develop/05_dApp/02_well-known-did-config.md (hunk 9 above)
Comment (lukeg90) [reply docs/develop/05_dApp/02_well-known-did-config.md]: SKYC does it ad-hoc

Comment (tjwelde) [reply docs/develop/05_dApp/02_well-known-did-config.md]: including self-attesting and putting it on-chain? Do I need to wait at least 12 seconds before I can connect to SKYC?

Comment (lukeg90) [reply docs/develop/05_dApp/02_well-known-did-config.md]: It's not on-chain

Comment (tjwelde) [reply docs/develop/05_dApp/02_well-known-did-config.md]: Got it, thanks

**Review by tjwelde — APPROVED** [review]
Looks pretty good. Let's go with it

**Pull request merged**